core/src/main/scala/org/apache/spark/util/ThreadUtils.scala

```diff
@@ -275,13 +274,7 @@ private[spark] object ThreadUtils {
    * @return new collection in which each element was given from the input collection `in` by
    *         applying the lambda function `f`.
    */
-  def parmap[I, O, Col[X] <: TraversableLike[X, Col[X]]]
-      (in: Col[I], prefix: String, maxThreads: Int)
-      (f: I => O)
-      (implicit
-        cbf: CanBuildFrom[Col[I], Future[O], Col[Future[O]]], // For in.map
-        cbf2: CanBuildFrom[Col[Future[O]], O, Col[O]] // for Future.sequence
-      ): Col[O] = {
+  def parmap[I, O](in: Seq[I], prefix: String, maxThreads: Int)(f: I => O): Seq[O] = {
```

Comment:
This signature is much less flexible. It always returns `Seq`. It forces us to use `asInstanceOf` like this:
```
val l = List(1, 2, 3)
l2 = parmap(l, ....)(...).asInstanceOf[List[...]]
```

Reply:
It seems that Apache Spark doesn't have those use cases yet. Was it for the future proof?

Reply:
Sure, but, do I necessarily have to get a List from a List? Seq -> Seq seems fine for all current usages.
That said, if there's any other way to write this that works in 2.12 and 2.13, definitely, open to that.

Reply:
@srowen I can give it a try. Does our build support 2.13 now?

Reply:
Heh, no, that's the hard part here. If you `./dev/change-scala-version.sh 2.13` after modifying it to accept that value, and tweak a few more things, and use `-Pscala-2.13` you can at least get to the point where there are still tons of compile errors including this one. There are a few things we can't fix now (i.e. deps that don't support 2.13) and are hard (i.e. things that will require multiple source trees). I'm fixing everything short of that that I can.

If you can write anything else in 2.12 that doesn't use TraversableLike, I can try it in my hacked up local build.

Reply:
This one works for 2.12: https://github.com/apache/spark/pull/26077 Could you try it with Scala 2.13?